### PR TITLE
Mqtt

### PR DESCRIPTION
**Pull request by aars**

### CODE REVIEW

**Review by Justin-Koningsberger — APPROVED** [review]
Lgtm!

File: expression-tracker/server.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'open3'
+require 'mqtt'
+require 'json'
+require_relative 'helper'
+include Helper
+
```
Comment (aars) [inline expression-tracker/server.rb:8]: This `mqtt` flag, the `mqtt_*` settings and the `--mqtt` option on line 14 are perfect candidates for the use of `OptionParser`. 

```ruby
require 'optparse'

@options = {
  mqtt: false,
  mqtt_broker: ENV["MQTT_BROKER"] || "mqtt://localhost",
  mqtt_topic: ENV["MQTT_TOPIC"]] || "PIT"
}
OptionParser.new do |opts|
  opts.banner = "Usage: server.rb [options]"

 opts.on("-m", "--mqtt", Boolean, "Use MQTT") do |i|
    @options[:mqtt] = true
 end
 opts.on("-b", "--mqtt_broker", String, "MQTT broker url") do |i|
    @options[:mqtt_broker] = i
 end
 opts.on("-b", "--mqtt_topic", String, "MQTT topic base") do |i|
    @options[:mqtt_topic] = i
 end
```

File: expression-tracker/server.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+@previous_values = {}
+
+# Simply tail output.csv and either publish on MQTT or print to STDOUT.
+Open3.popen3('tail -f processed/output.csv') do |_stdin, stdout, _stderr, _wait_thr|
+	stdout.each_line do |line|
+		@values = {}
+		line.split(',').map(&:to_f).each_with_index { |n, i| @values[headers[i]] = n }
+
+		reduced      = Hash[output_headers.map { |n| [headers[n], @values[headers[n]]] }]
+		reduced_line = JSON[reduced]
```
Comment (aars) [inline expression-tracker/server.rb:40]: We could just wrap line 38 in `JSON[]`, but this is where we can switch output format if we need something other than JSON, perhaps XML, YAML, TOML or SuperX++. We could also switch this out for a `format()` function that does some pre-processing of the selected values before sending it out.

File: expression-tracker/server.rb (hunk 1 above)
Comment (aars) [reply expression-tracker/server.rb:8]: @Justin-Koningsberger Dropped some tips for ya! Cheers.

Comment (Justin-Koningsberger) [reply expression-tracker/server.rb:8]: Seems, like a good solution, I wasn't aware of OptionParser.

File: expression-tracker/server.rb (hunk 2 above)
Comment (Justin-Koningsberger) [reply expression-tracker/server.rb:40]: The `format()` function would be my preference, seems cleanest. Maybe switch output format there, if certain opts are on/set.

**Pull request merged**